src/Step/Deploy/SplitDbConnection.php

```diff
@@ -110,56 +110,60 @@ public function __construct(
      */
     public function execute()
     {
-        if ($this->flagManager->exists(FlagManager::FLAG_IGNORE_SPLIT_DB)) {
-            $this->logger->info(sprintf(
-                'Enabling a split database will be skipped. The flag %s was detected.',
-                FlagManager::FLAG_IGNORE_SPLIT_DB
-            ));
-            $this->flagManager->delete(FlagManager::FLAG_IGNORE_SPLIT_DB);
-            return;
-        }
+        try {
+            if ($this->flagManager->exists(FlagManager::FLAG_IGNORE_SPLIT_DB)) {
+                $this->logger->info(sprintf(
+                    'Enabling a split database will be skipped. The flag %s was detected.',
+                    FlagManager::FLAG_IGNORE_SPLIT_DB
+                ));
+                $this->flagManager->delete(FlagManager::FLAG_IGNORE_SPLIT_DB);
+                return;
+            }
 
-        $splitTypes = $this->stageConfig->get(DeployInterface::VAR_SPLIT_DB);
+            $splitTypes = $this->stageConfig->get(DeployInterface::VAR_SPLIT_DB);
 
-        $dbConfig = $this->dbConfig->get();
-        $notAvailableSplitTypes = $this->getMissedSplitTypes(
-            $splitTypes,
-            $dbConfig[DbConfig::KEY_CONNECTION] ?? []
-        );
+            $dbConfig = $this->dbConfig->get();
+            $notAvailableSplitTypes = $this->getMissedSplitTypes(
+                $splitTypes,
+                $dbConfig[DbConfig::KEY_CONNECTION] ?? []
+            );
 
-        if (!empty($notAvailableSplitTypes)) {
-            $this->logger->warning(
+            if (!empty($notAvailableSplitTypes)) {
+                $this->logger->warning(
                 sprintf(
```

Comment:
Indent of lines 133/137/138 causing tests to fail - should be indented a further 4 spaces.

Reply:
fixed

Reply:
@BarnyShergold please approve. Requested changes were implemented